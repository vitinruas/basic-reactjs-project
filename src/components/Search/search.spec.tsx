import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Search } from '.'

const fakeProps = {
  value: 'any_value',
  handleOnChange: jest.fn(),
}

interface IHTMLElement extends HTMLElement {
  value: any
}
describe('<Search />', () => {
  test('Should exists a value', () => {
    render(<Search {...fakeProps} />)

    const input = screen.getByPlaceholderText(/type to look for something/i)

    expect((input as IHTMLElement).value).toBe('any_value')
  })
  test('Should call handleChange function on each key pressed', () => {
    render(<Search value="" handleOnChange={fakeProps.handleOnChange} />)

    const input = screen.getByPlaceholderText(/type to look for something/i)
    const value = 'value_typed'
    userEvent.type(input, value)

    expect(fakeProps.handleOnChange).toHaveBeenCalledTimes(value.length)
  })
})
