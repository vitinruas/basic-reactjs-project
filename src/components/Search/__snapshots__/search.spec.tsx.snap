// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match a snapshot 1`] = `
<div
  class="search-container"
>
  <span
    class="search-title"
  >
    Searching for: 
    <strong>
      any_value
    </strong>
  </span>
  <input
    class="search-input"
    placeholder="Type to look for something"
    type="text"
    value="any_value"
  />
</div>
`;
